src/main/java/io/leangen/graphql/annotations/GraphQLIgnore.java

```diff
@@ -10,6 +10,6 @@
  * by the mapping process
  */
 @Retention(RetentionPolicy.RUNTIME)
-@Target({ElementType.ANNOTATION_TYPE, ElementType.METHOD, ElementType.FIELD})
+@Target({ElementType.TYPE, ElementType.ANNOTATION_TYPE, ElementType.METHOD, ElementType.FIELD})
```

Comment:
`ElementType.ANNOTATION_TYPE` is redundant in this case, as `ElementType.TYPE` already covers it.
But... could you please separate this into a different PR as it pertains to a different issue... Also, I didn't fully decide this is the right approach. It's very ambiguous what is being ignored here. The entire class is marked, yet it is only ignored in one very narrow scenario. I'm not sure that's a good design. I have to think more about that one before implementing it.

Reply:
Yes, ElementType.ANNOTATION_TYPE is redundant, shall I correct it and push, see my comment below.

You are right in the sense of clean design, but having this feature will give the flexibility to ignore a sub type if the developer wants to hide it from a public api (Given that the developer can't design the schema but he follows some open standard, in our case we are developing a FHIR GraphQL API https://www.hl7.org/fhir/overview.html by mapping FHIR JSON Schema to GraphQL )
When you want to develop a GraphQL application, usually there are two ways, code first and schema first. There is another way, that is, you have an external schema (like FHIR) and you start coding domain classes by following the external schema. Since this framework is not about opinions but flexibly, i feel like it will be useful to someone.

Reply:
I was thinking a lot about the design of this one, but can't seem to come up with anything satisfactory... So for now at least, please correct the redundant target and I'll merge it as-is.